### DISCUSSION TITLE
Using PHP to send FCM notification getting error "error":"MismatchSenderId"

### DISCUSSION THREAD

**Discussion by anilkumar2273702** [invertase/react-native-firebase#5385] [Q&A] [answered]
I'm using PHP to send FCM notification to Expo android apps, but getting error - "error":"MismatchSenderId" 

![image (2)](https://user-images.githubusercontent.com/66350452/120497749-4c953380-c3dc-11eb-913b-eabbd2617158.png)

**Comment by mikehardy** [comment, marked as answer]
Sounds like you should match your sender id.

**Reply by anilkumar2273702** [reply]
@mikehardy Where should I match sender id in PHP code? 
I'm using the following code in PHP - 


$regId ="egDtmlLURSqb__yBaAAQHt:APA91bGfFVzyggKG-IGZc8zCUCGtEl95Gll6N3zbNAy5........................"; // "REGISTRATION ID" from the response above. If this is wrong it throws an error ("InvalidRegistration"), so I know that this is correct.


$url = 'https://fcm.googleapis.com/fcm/send';
// $msg = array
//                         (
//                         'body'  => 'body msg',
//                         'title' => 'title',
//                         'icon'  => 'myicon',/*Default Icon*/
//                         'sound' => 'mySound'/*Default sound*/
//                         );

$fields = array (
        'registration_ids' => array (
                $regId
        ),
        'data' => array (
                 "message" => "Test"
        )
       // 'to'            =>  $regId,
       // 'notification'  => $msg
);
$fields = json_encode ( $fields );

$headers = array (
        'Authorization: key=' . "AAAA792FeiQ:APA91bGcnMy8I2mzEDO5Z98WTvMvHMlJDUdO0Ood-_xFdF1NDQ2mKwyhyPYrSSysoX0EqQdp71gU-.........................", //This is the "Server key" above "Sender ID"
        //This matches "Project Settings" > "Cloud Messaging" > [Server Key]
        //If this is wrong it returns: INVALID_KEY error 401. So I know this is correct.
        'Content-Type: application/json'
);

$ch = curl_init ();
curl_setopt ( $ch, CURLOPT_URL, $url );
curl_setopt ( $ch, CURLOPT_POST, true );
curl_setopt ( $ch, CURLOPT_HTTPHEADER, $headers );
curl_setopt ( $ch, CURLOPT_RETURNTRANSFER, true );
curl_setopt ( $ch, CURLOPT_POSTFIELDS, $fields );

$result = curl_exec ( $ch );
echo $result;
curl_close ( $ch );

**Reply by mikehardy** [reply]
This seems better suited to stackoverflow
Your problem will not result in a change to this module, this module works for sending and receiving FCM so what you have is a project-specific problem and stackoverflow will get you answers from the wider community
